src/main/java/org/folio/circulation/domain/notice/schedule/DueDateNotRealTimeScheduledNoticeHandler.java

```diff
@@ -63,14 +64,23 @@ public DueDateNotRealTimeScheduledNoticeHandler(
   }
 
   private CompletableFuture<Result<Void>> handleNoticeGroup(List<ScheduledNotice> noticeGroup) {
-  return allOf(noticeGroup, this::getContext)
+  return allResultsOf(noticeGroup, this::getContext)
```

Comment:
If I correctly understand, the removal logic works in two steps:
1. `getContext` removes a notice if a reference record missed (either loan, item or user);
1. Then `handleNoticeGroup` method invokes `handleFailures` method that removes the failed notices from the list, so they won't be sent.

Am I correct understand?

If so, then I think we should do these two operations within single place/method, as they actually represent a single operation.

Reply:
You did understand it correctly. But in this case even though these are two steps of the same operation, they cannot be done in one place: each invocation of `getContext` has to return something even for an "invalid" notice. So we still have to handle and filter all the results in `handleNoticeGroup`.

Reply:
You did understand it correctly. But in this case even though these are two steps of the same operation, they cannot be done in one place: each invocation of `getContext` has to return something even for an "invalid" notice. So we still have to handle and filter all the results in `handleNoticeGroup`.

Reply:
Ok, let's leave it as is.